src/IntegrationUtils.ts

```diff
@@ -181,9 +201,52 @@ function getSelectedResource() {
 		}
 	});
 }
+		
+function getSelectedProperties() {
+	return new Promise <any> ( async (resolve, reject) => {
+		let hasMoreProperties: boolean = true;
+		let returnedProperties : string[] = [];
+		while (hasMoreProperties) {
+			let newProperty = undefined;
+			await vscode.window.showInputBox({
+				placeHolder: 'Specify the property name',
+				ignoreFocusOut: true,
+				validateInput: validateName
+			}).then ( async (propName) => {
+				await vscode.window.showInputBox({
+					placeHolder: 'Specify the property value',
+					ignoreFocusOut: true
+				}).then ( (propValue) => {
+					if (propValue) {
+						propValue = propValue.replace(/"/g, '\\"');
+						newProperty = `${propName}="${propValue}"`;
+					}
+				});
+			});
+			if (!newProperty) {
+				reject(new Error(`No Property provided`));
+				return;
+			} else {
+				const answer = await vscode.window.showQuickPick( ['No', 'Yes'], {
+					placeHolder: 'Are there more properties?',
+					canPickMany : false	});
+				if (!answer) {
+					reject(new Error(`No Property answer given`));
+					return;
+				}
+				returnedProperties.push(newProperty);
+				if (answer === 'No') {
+					hasMoreProperties = false;
+					resolve(returnedProperties);
+					return;
```

Comment:
obsolete return statement

Reply:
See my earlier comment about reject/return

Reply:
there is no more code to execute after that line.

Reply:
I'd rather be consistent than not

Reply:
Cleaned up